### DISCUSSION TITLE
How do I view print() output from plugin's main.lua ?

### DISCUSSION THREAD

**Discussion by tn777Terp** [xournalpp/xournalpp#5100] [Q&A] [answered]
OS: Win10
Installed Lua Version: 5.4.2

Hi. I would like to make a plugin because I don't want to have to set up and run a build. However, I don't know where I can view the outputs of the print() statements for debugging-- I'm talking about the print statement in main.lua. I tried running Xournal from cmd prompt hoping that I can view the output there, but nothing is printing out.

My workaround right now is to have the output load a app.msgbox and print out the variable values there, but this seems awfully inefficient.

Is there a specific tool or steps I need to take to get the lua prompt to print out? I have Lua 5.4.2 installed.

**Comment by rolandlo** [comment, marked as answer]
On Windows you can use the MinGW64-console that you get when installing [MSYS2](https://www.msys2.org/). In the cmd prompt you don't get console messages.

By the way: if you use the nightly build, `app.msgbox` has recently been deprecated and replaced by `app.openDialog` (see https://github.com/xournalpp/xournalpp/blob/master/src/core/plugin/luapi_application.h). It will continue to work for a while, but eventually be removed.
